Bluetooth: Kconfig menu cleanup

rugeGerritsen opened the pull request:
The number of options presented to the user is becoming quite large, which can be quite intimidating at first. By grouping them it makes it easier to get an overview of what can be configured.

rugeGerritsen committed:
Bluetooth: Kconfig: Move Host configuration to a separate menu

This reduces the number of visible options.

Signed-off-by: Rubin Gerritsen <rubin.gerritsen@nordicsemi.no>

rugeGerritsen committed:
Bluetooth: Kconfig: Move GATT Kconfig to be part of host

This prevents GATT service selection being selectable when compiling
controller only builds.

Signed-off-by: Rubin Gerritsen <rubin.gerritsen@nordicsemi.no>

rugeGerritsen committed:
Bluetooth: Kconfig: Remove Controller comment line

The controller configuration is already a submenu

Signed-off-by: Rubin Gerritsen <rubin.gerritsen@nordicsemi.no>

rugeGerritsen committed:
Bluetooth: Kconfig: No use menu for observer and broadcaster configs

This will ensure that they don't turn up as sub-menus

Signed-off-by: Rubin Gerritsen <rubin.gerritsen@nordicsemi.no>

rugeGerritsen committed:
Bluetooth: Kconfig: Move BT_LIM_ADV_TIMEOUT to host config

This configuration is meant to be in the host.

Signed-off-by: Rubin Gerritsen <rubin.gerritsen@nordicsemi.no>

rugeGerritsen committed:
Bluetooth: Kconfig: Hide buffer configuration in a separate menu

Users will usually not want to browse through these. Therefore it is
better to hide them a bit.

Signed-off-by: Rubin Gerritsen <rubin.gerritsen@nordicsemi.no>

Thalley commented on subsys/bluetooth/host/Kconfig:
```diff
@@ -441,6 +441,7 @@ endif # BT_SMP
 
 rsource "Kconfig.l2cap"
 rsource "Kconfig.gatt"
+rsource "../services/Kconfig"
```
If we are moving services to the host Kconfig, should we do the same for `mesh` and `audio`?

rugeGerritsen replied:
That is a good point! I can do so if others agree

Thalley replied:
I don't have the correct answer, but IMO `services` should be at the same level as `mesh` and `audio` given the subsystem structure. So if we move `services` from the common Kconfig to the host, then I think `mesh` and `audio` should too. 

That being said, `services` can also stay where it is, as that reflects the subsystem structure. 

@jhedberg Any input?

jhedberg replied:
AFAIK stuff like mesh, audio and services, even though logically layers above the host, were moved under `subsys/bluetooth` to avoid having super deep directory hierarchies. However yes, when it comes to Kconfig enablement, they should probably all depend on the host being enabled first and not reside somewhere higher up in the selection tree.

Thalley replied:
Let's move the `audio` and `mesh` `rsource` lines to the same place as `services` then :)

rugeGerritsen replied:
Adding audio resulted in some cyclic dependencies. Looking into it now

asbjornsabo replied:
I am not sure I see what the difference is between those Kconfigs now being rsourced from susbys/bluetooth/host/Kconfig, and those being rsourced from subsys/bluetooth/Kconfig

Should e.g. Kconfig.iso and Kconfig.adv also be rsourced here?
(Is adv by the way considered a part of the host?  It is in the spec.)

rugeGerritsen replied:
I wanted to ensure those configs are not visible when performing a controller build. AFAIK the Kconfig.adv and Kconfig.iso contain global stack configuration options similar to `subsys/bluetooth/Kconfig`. Therefore, these cannot be guarded by the host.

asbjornsabo replied:
Maybe further refactoring is in place then - to split those into host, controller and common parts?

Also wondering the ways to do dependencies - rsourcing from within dependency as here, vs. select/depends.
Anyway - this PR is still an improvement to the current situation.

Thalley replied:
Features that are common between the host and controller (e.g. adv and iso) makes sense to keep as common configs imo.

rugeGerritsen committed:
Bluetooth: Kconfig: Move audio and mesh to be shown under host

These subsystems are part of the host.

Signed-off-by: Rubin Gerritsen <rubin.gerritsen@nordicsemi.no>

rugeGerritsen commented:
Rebased on main. Updated the last commit ( https://github.com/zephyrproject-rtos/zephyr/pull/42852/commits/f020389a06ba4b181f10a4528614ae1e1b89093e) to ensure that mesh and audio didn't add dependencies to BT_CONN.

asbjornsabo reviewed (approved):
LGTM
